src/game/gui/textslider.c

```diff
@@ -16,35 +16,37 @@ typedef struct {
     int *pos;
     int has_off;
     int positions;
+    char txt[20];
 
     void *userdata;
     textslider_slide_cb slide;
 } textslider;
 
-static void textslider_render(component *c) {
+static void render_text(component *c) {
     textslider *tb = widget_get_obj(c);
-    str txt;
-    str_from_format(&txt, "%s ", tb->text);
+    int off = snprintf(tb->txt, sizeof(tb->txt), "%s ", tb->text);
     if(tb->has_off && *tb->pos == 0) {
-        str_append_c(&txt, "OFF");
+        snprintf(tb->txt + off, sizeof(tb->txt) - off, "%s", "OFF");
```

Comment:
I dislike this -- the idea with str type is that we can later convert it to utf-8. This is taking steps backwards. How about we cache the str to textslider instead ?